### PR TITLE
mgr/dashboard: Resolve TestBed performance issue

### PR DESCRIPTION
**Pull request by Devp00l**
With this helper function you can easily resolve the TestBed resetting
performance issue. If more tests exists in a test suite, it makes sense
to configure TestBed only once if you are not doing a lot of TestBed
specific stuff (haven't hit the limitation). It will reduce the test
run time by around $tests * 50 %. In my case it was a test suite with
47 tests with a run time of over 30s after using the static test bed
method it ran in 1.2s. The run time was reduced to 0.04 %! This is
equivalent to a speed increase of 2500% (100/0.04)!

For our own security the normal way will be taken if you not
set the _DEV_ configuration variable to true. It will be false when
"run-frontend-unittests.sh" is run.

Signed-off-by: Stephan Müller <smueller@suse.com>

### CODE REVIEW

**Comment by Devp00l** [comment]
There is a compiling issue that I will fix the compiling issue tomorrow.

**Comment by Devp00l** [comment]
Fixed the compiling issue :)

**Comment by tspmelo** [comment]
Is this used automatically or needs to be imported in each spec file?

**Comment by ricardoasmarques** [comment]
>  In my case it was a test suite with...

Can you share the test suite where you are using this hack/workaround?

**Comment by Devp00l** [comment]
@tspmelo You have to import it manually if you want to use it

@ricardoasmarques it is used in the pool-form spec file of my WIP pool form PR it looks like this:
```
  unitTestHelper.staticTestBed({
    declarations: [ PoolFormComponent ],
    imports: [
      ReactiveFormsModule,
      SharedModule,
      HttpClientTestingModule,
      RouterTestingModule,
      ToastModule.forRoot(),
      FormsModule
    ],
    providers: [
      ErasureCodeProfileService,
      {provide: ActivatedRoute, useValue: {params: Observable.of({name: 'somePoolName'})}}
    ]
  });
```
And that's all the magic you have to do to get the unit tests to a normal unit test speed :)

**Comment by LenzGr** [comment]
Closing this one in favor of #22031, which seems to be a more promising approach.

**Comment by Devp00l** [comment]
#22031 won't help as jest will also run the whole suite on file change or just rerun all previous failed tests. This will not resolve the issue. Also this PR doesn't change how the tests a run - you have to explicitly enable it's usage. This way you test the real thing, the other solution to increase the speed would be to mock everything, this will reduce the tested code and you get a mess of mocks inside the test suite.

The performance issue was introduced by the vendor, if it's fixed it is a easy thing to remove it. Unit tests are mend to be fast not as slow as they currently are through this performance issue.

BTW the suite needs 95s to run with jest now instead of 2s with this solution.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/unit-test-helper.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { async, TestBed } from '@angular/core/testing';
+
+import { _DEV_ } from '../../unit-test-configuration';
+
+export class UnitTestHelper {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/unit-test-helper.ts]: You could export a static class or simply export the method, that way you don't have to initialize it every time on the spec files.

**Comment by Devp00l** [comment]
Addressed all changes

**Comment by Devp00l** [comment]
Addressed all comments and resolved all conflicts

**Comment by tspmelo** [comment]
`clients.component.spec.ts` and `cephfs.component.spec.ts` were removed in a previous commit.
They should not be included in the PR.

**Comment by Devp00l** [comment]
Addressed all comments :)

**Comment by tspmelo** [comment]
Could you remove the 2 commits that modify tests from this PR?
That way we can discuss the other changes in another PR and merge this one.

**Comment by Devp00l** [comment]
@tspmelo Split the PR https://github.com/ceph/ceph/pull/22473 and added missing files

**Pull request merged**